content/docs/reference/design-docs/stdcm/pathfinding_module/implementation_details.en.md

```diff
@@ -0,0 +1,68 @@
+---
+title: "Implementation details"
+linkTitle: "6 - Implementation details"
+weight: 60
+---
+
+
+This page is about implementation details.
+It isn't necessary to understand general principles,
+but it helps before reading the code.
+
+#### STDCMEdgeBuilder
+
+This class is used to make it easier to create instances of
+`STDCMEdge`, the graph edges. Those contain many attributes,
+most of which can be determined from the context (e.g. the
+previous node).
+The `STDCMEdgeBuilder` class makes some parameters optional
+and automatically computes others.
+
+Once instantiated and parameterd, an `STDCMEdgeBuilder` has two methods:
+
+
+- `Collection<STDCMEdge> makeAllEdges()` can be used to create all
+the possible edges in the given context for a given route.
+If there are several "openings" between occupancy blocks, one edge
+is instantiated for each opening. Every conflict, their avoidance,
+and their related attributes are handled here.
+
+- `STDCMEdge findEdgeSameNextOccupancy(double timeNextOccupancy)`:
+This method is used to get the specific edges that uses a certain
+opening (when it exists), identified here with the time of the next
+occupancy block. It is called whenever a new edge must be re-created
+to replace an old one. It calls the previous method.
+
+
+### Pathfinding
+
+#### Cost function
+
+The function used to define pathfinding cost sets which path
+is used over another. The result is always the one that minimizes
+this cost (as long as the heuristic is admissible).
+
+Here, two parameters are used: total run time and departure time.
+The latter has a very small weight compared to the former,
+so that the fastest path is found. More details
+are explained in the documentation of those methods.
```

Comment:
```suggestion
so that the fastest path is found. More details
can be found in methods documentation <LINK>.
```

Reply:
Ideally we'd link the javadoc, but I don't know if/how we can do that. There's a javadoc file on the website but I don't know how to use it.

I'll link the .java file on github for now.